src/nodes/nodes.go

```diff
@@ -0,0 +1,171 @@
+package nodes
+
+type Node interface {
+	GetLineno() int
+}
+
+type NodeCommon struct {
+	Lineno int
+}
+
+func (n *NodeCommon) GetLineno() int {
+	return n.Lineno
+}
+
+var _ Node = &NodeCommon{}
+
+type Template struct {
+	Body []Node
+	NodeCommon
+}
+
+type Stmt interface {
+}
+
+type Expr interface {
+}
+
+type Output struct {
+	Nodes []Node
+	NodeCommon
+}
+
+// TemplateData represents a constant template string.
+type TemplateData struct {
+	Data string
+	NodeCommon
+}
+
+type Tuple struct {
+	Items []Node
+	Ctx   string
+	NodeCommon
+}
+
+type Const struct {
+	Value any
+	NodeCommon
+}
+
+type Name struct {
+	Name string
+	Ctx  string
+	NodeCommon
+}
+
+type CondExpr struct {
+	Test  Node
+	Expr1 Node
+	Expr2 Node
+	NodeCommon
+}
+
+type Or struct {
```

Comment:
Why arent logic nodes also BinExpr or UnaryExpr?

Reply:
I converted them to BinOp and UnaryOp, but a thing to note is that they don't have defined tokens in lexer (I've added comments in nodes.go)